### PR TITLE
Fix Manhwa18

### LINKED ISSUE #5996: Manhwa18: error 404
### Source information

Manhwa18 1.4.11

### Source language

English

### Steps to reproduce

Access extension

### Expected behavior

List of manhwa should appear

### Actual behavior

http 404

### Mihon/Tachiyomi version

0.17.0

### Android version

Android 10

### Other details

url of this website still the same, I can also open it in webview so seem like logic is changed

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by cuong-tran**
Migrating to new theme

Closes #5996 

Checklist:

- [X] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [X] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio
- [X] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt [hunk 1]
```diff
@@ -1,62 +1,217 @@
...
+class Manhwa18 : HttpSource() {
+
+    override val baseUrl = "https://manhwa18.com"
+    private val apiUrl = "https://cdn3.manhwa18.com/api/v1"
+    override val lang = "en"
+    override val name = "Manhwa18"
+    override val supportsLatest = true
 
-    // Migrated from FMReader to MyMangaCMS.
     override val versionId = 2
```
Comment (AwkwardPeak7) [inline src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt:30]: bump if urls have changed

File: src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt [hunk 2]
```diff
@@ -1,62 +1,217 @@
...
+                    }
+                    else -> {}
+                }
+            }
+        }
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET("$apiUrl/get-search-suggest/$query", headers)
```
Comment (AwkwardPeak7) [inline src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt]: use `HttpUrl.Builder` to properly encode the query

File: src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt [hunk 3]
```diff
@@ -1,62 +1,217 @@
...
+        val mangaDetail = json.decodeFromString<MangaDetail>(response.body.string())
+        val mangaSlug = mangaDetail.manga.slug
+
+        return mangaDetail.manga.episodes?.map { chapter ->
+            SChapter.create().apply {
+                // compatible with old theme
+                setUrlWithoutDomain("/manga/$mangaSlug/${chapter.slug}")
+                name = chapter.name
+                date_upload = chapter.created_at?.parseDate() ?: 0L
+                chapter_number = chapter.name.toFloat()
```
Comment (AwkwardPeak7) [inline src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt]: app will try to set the number itself, check if that one works correctly for this source

File: src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18Dto.kt [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    @SerialName("data") val mangaList: List<Manga>,
+)
+
+@Serializable
+data class MangaDetail(
+    @SerialName("product") val manga: Manga,
+)
+
+@Serializable
+data class Manga(
```
Comment (AwkwardPeak7) [inline src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18Dto.kt:25]: remove unused fields

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt (hunk 1 above)
Comment (cuong-tran) [reply src/en/manhwa18/src/eu/kanade/tachiyomi/extension/en/manhwa18/Manhwa18.kt:30]: the manga/chapter url stay the same, no need to update version so user can continue using existed library entries without needs to migrate

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**